ui/app/AppLayouts/Communities/views/CommunityTokenView.qml

```diff
@@ -196,27 +203,108 @@ StatusScrollView {
+                onViewMessagesRequested: root.viewMessagesRequested(contactId)
+                onAirdropRequested: root.airdropRequested(address)
+                onGeneralAirdropRequested: root.generalAirdropRequested()
+                onRemoteDestructRequested: root.remoteDestructRequested(name, address)
+
+                onKickRequested: root.kickRequested(name, contactId, address)
+                onBanRequested: root.banRequested(name, contactId, address)
+            }
+        }
+
+        Component {
+            id: tokenHolderContact
 
-            onViewProfileRequested: root.viewProfileRequested(contactId)
-            onViewMessagesRequested: root.viewMessagesRequested(contactId)
-            onAirdropRequested: root.airdropRequested(address)
-            onGeneralAirdropRequested: root.generalAirdropRequested()
-            onRemoteDestructRequested: root.remoteDestructRequested(name, address)
+            ColumnLayout {
+                Layout.topMargin: Style.current.padding
+                Layout.fillWidth: true
 
-            onKickRequested: root.kickRequested(name, contactId, address)
-            onBanRequested: root.banRequested(name, contactId, address)
+                StatusBaseText {
+                    Layout.fillWidth: true
+
+                    wrapMode: Text.Wrap
+                    font.pixelSize: Style.current.primaryTextFontSize
+                    color: Theme.palette.baseColor1
+
+                    text: qsTr("%1 token hodler").arg(root.name)
+                }
+
+                StatusInfoBoxPanel {
+                    id: infoBoxPanel
+
+                    Layout.fillWidth: true
+                    Layout.topMargin: Style.current.padding
+
+                    enabled: root.deploymentCompleted && (token.infiniteSupply || token.remainingTokens > 0)
+                    visible: root.tokenOwnersModel.count === 0
+                    title: qsTr("No hodlers just yet")
+                    text: qsTr("You can Airdrop tokens to deserving Community members or to give individuals token-based permissions.")
+                    buttonText: qsTr("Airdrop")
+
+                    onClicked: root.generalAirdropRequested()
+                }
+
+                StatusListView {
+                    id: tokenOwnerList
+                    objectName: "tokenOwnerList"
+
+                    clip: false
+
+                    Layout.fillWidth: true
+                    Layout.fillHeight: true
+                    anchors.bottomMargin: Style.current.bigPadding
+                    displayMarginEnd: anchors.bottomMargin
+
+                    model: root.tokenOwnersModel
+                    delegate: StatusMemberListItem {
+                        color: "transparent"
+                        leftPadding: 0
+                        rightPadding: 0
+                        sensor.enabled: false
+
+                        Component.onCompleted: {
+                            d.updateContactDetails(model.contactId)
```

Comment:
Hmm, how does this work? You're fetching and overwriting the `d.contactDetails` for each delegate in the listview? :thinking: That sounds odd...

Reply:
From the tokenOwnersModel, what could be extracted is the name and the contactId (cf. below. The other way to go about this would be to enrich the `tokenOwnersModel` with all the necessary details to be displayed and then fetch the information directly from the tokenOwners, not sure if this second approach is overkill.

> Hmm, how does this work? You're fetching and overwriting the d.contactDetails for each delegate in the listview?
As of now, There is only 1 Owner token for a community so, the delegate is called once.

```
  TokenOwnersItem* = object
    contactId*: string
    name*: string
    imageSource*: string
    numberOfMessages*: int
    ownerDetails*: CollectibleOwner
    amount*: Uint256
    remotelyDestructState*: ContractTransactionStatus
```

Reply:
The solution would be to use the `LeftJoinModel` on the `contactId`

Reply:
To accomplish what is asked by @caybro 
I should pull the usersModel and  Join it witht he tokenOwnersModel on the contactId because it's better to use the Qt binding rather than fetching the contact details even if it's called only once here as of now (I suspect it might be called more than once later on). This is ideal solution for sure, it might take me a bit more time than I had expected. I'll be working on it now

Reply:
So I have cooked a new revision making use of a LeftJoinModel for extracting the ownerToken after crossing it with the members available to the CommunitySettingsView component. Please @caybro let me know what you think